src/workflows/trades.ts

```diff
@@ -56,12 +56,11 @@ export async function createTradeWorkflow(
 }
 
 export async function createTradeWorkflowWithSigner(
-  l1Signer: L1Signer,
-  l2Signer: L2Signer,
+  walletConnection: WalletConnection,
   request: GetSignableTradeRequest,
   tradesApi: TradesApi,
 ) {
-  const ethAddress = await l1Signer.getAddress();
+  const ethAddress = await walletConnection.l1Signer.getAddress();
```

Comment:
```suggestion
  const {l1Signer, l2Signer} = walletConnection;
  const ethAddress = await l1Signer.getAddress();
```

and you can remove all the other walletConnection.l1Signer, l2Signer as well.